composer/core/time.py

```diff
@@ -367,14 +367,21 @@ class Timer(Serializable):
         A user need not instantiate this class.
     """
 
-    def __init__(self):
-        self._epoch = Time(0, TimeUnit.EPOCH)
-        self._batch = Time(0, TimeUnit.BATCH)
-        self._sample = Time(0, TimeUnit.SAMPLE)
-        self._token = Time(0, TimeUnit.TOKEN)
-        self._batch_in_epoch = Time(0, TimeUnit.BATCH)
-        self._sample_in_epoch = Time(0, TimeUnit.SAMPLE)
-        self._token_in_epoch = Time(0, TimeUnit.TOKEN)
+    def __init__(self,
+                 epoch: Time = Time(0, TimeUnit.EPOCH),
```

Comment:
Would it be possible to remove the default values? While 0 is a reasonable default (most of the time), I think it would be preferable for all arguments to be explicitly specified (similar to how the current Timestamp class works). I am mainly concerned about the case where we attempt to create a new timestamp from an existing timestamp, but forget to copy over one of the fields.

Reply:
Sure, thanks for providing the rationale. I will try and implement your suggestions.

Reply:
After further thought, I left the default values here, but added a `copy` method to get a copy of the timestamp with some values changed. That will handle the case where we create a new timestamp from the existing one, and only want to update some of the fields.